ScipDotnet/ScipSymbolFormatter.cs

```diff
@@ -0,0 +1,376 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.Extensions.Logging;
+using Scip;
+using Document = Scip.Document;
+
+namespace ScipDotnet;
+
+/// <summary>
+/// Creates SCIP <code>Document</code> based on provided symbols.
+/// </summary>
+public class ScipSymbolFormatter
```

Comment:
My initial interpretation of the name `ScipSymbolFormatter` was that it formats SCIP symbols like this interface we have in Go https://sourcegraph.com/github.com/sourcegraph/scip/-/blob/bindings/go/scip/symbol_formatter.go?L10

I propose we rename this class into something else to make it clearer that this class does more than just format symbols

```suggestion
public class ScipDocumentIndexer
```

I also propose we rename `ScipIndexer` into `ScipProjectIndexer`